packages/@lwc/integration-karma/test/api/CustomElementConstructor-getter/index.spec.js

```diff
@@ -25,6 +18,13 @@ it('should throw when trying to claim abstract LightningElement as custom elemen
 });
 
 if (SUPPORTS_CUSTOM_ELEMENTS) {
+    it('should return a custom element', () => {
+        class Test extends LightningElement {}
+        const TestCustomElement = Test.CustomElementConstructor;
+
+        expect(typeof TestCustomElement).toBe('function');
+    });
+
```

Comment:
This throws an error in IE11 due to this line:

https://github.com/salesforce/lwc/blob/77e9d86b190b7c60e5c825322f8eeeb0acc755e1/packages/%40lwc/engine-dom/src/apis/build-custom-element-constructor.ts#L59

Something about how Babel transforms this causes it to become an error in IE11. I didn't think it was important enough to fix, since `CustomElementConstructor` could only work in IE11 with a lot of polyfills anyway, and none of the other tests confirmed that `CustomElementConstructor` even worked in IE11.

Reply:
This is an observable change. Here are my thoughts about the impact.

Does this work in other non-IE11 but COMPAT browsers?
```
Chrome | < 56
Firefox | < 53
Edge | < 15
Safari | < 11
```

- If it is only failing in IE11, then it's still okay IMO. I assume nobody is trying access this getter in IE11 because it has no native custom element support.

- If it fails in other COMPAT browsers, for example Safari 10.1, then its a noticeable observable change.

It is unfortunate that there was no guarding for this API to make sure the getter never worked in COMPAT mode(specifically when native custom elements is not supported). Can we please add an non-prod assertion in `lightning-element.ts - getCustomElementConstructor` to verify that it is not invoked if `hasCustomElements` is false.

Reply:
Please see the [observable change analysis doc](https://salesforce.quip.com/onyJANMUpGdl).

In any browser that doesn't support custom elements, `new Ctor.CustomElementConstructor()` will throw an error. The fact that calling `Ctor.CustomElementConstructor` does _not_ throw an error is kind of an edge case IMO.

For this to be a problem, we would have to assume that someone is supporting older browsers that don't support custom elements, they are using `Ctor.CustomElementConstructor`, but they are not `new`ing the constructor. This seems like a really odd scenario to me. 🙂 

FWIW I don't recall perfectly, but I believe this was only throwing in IE11, not in other compat browsers.